Sources/NIOPosix/Bootstrap.swift

```diff
@@ -521,6 +521,200 @@ public final class ServerBootstrap {
+        ///                   Must be a valid filesystem path and should not exist 
+        ///                   unless cleanup is enabled in the binding operation
+        /// - Warning: The path must not exist.
+        public static func unixDomainSocketPath(_ path: String) -> BindTarget {
+            BindTarget(base: .unixDomainSocketPath(path))
+        }
+
+        /// Creates a binding target for a VSOCK address.
+        ///
+        /// VSOCK (Virtual Socket) provides communication between virtual machines and their hosts,
+        /// or between different virtual machines on the same host. This is commonly used
+        /// in virtualized environments for guest-host communication.
+        ///
+        /// - Parameter vsockAddress: The VSOCK address to bind to, containing both 
+        ///                          context ID (CID) and port number
+        /// - Note: VSOCK support depends on the underlying platform and virtualization technology
+        public static func vsock(_ vsockAddress: VsockAddress) -> BindTarget {
+            BindTarget(base: .vsock(vsockAddress))
+        }
+
+        /// Creates a binding target for an existing socket handle.
+        ///
+        /// This method allows you to use a pre-existing socket that has already been
+        /// created and optionally configured. This is useful for advanced scenarios where you
+        /// need custom socket setup before binding, or when integrating with external libraries.
+        ///
+        /// - Parameters:
+        ///   - handle: The existing socket handle to use. Must be a valid, open socket
+        ///            that is compatible with the intended server bootstrap type
+        /// - Note: The bootstrap will take ownership of the socket handle and will close
+        ///         it when the server shuts down
+        public static func socket(_ handle: NIOBSDSocket.Handle) -> BindTarget {
+            BindTarget(base: .socket(handle))
+        }
+    }
+
+    /// Bind the `ServerSocketChannel` to the ``BindTarget``. This method will returns once all connections that
+    /// were spawned have been closed.
+    ///
+    /// # Supporting graceful shutdown
+    ///
+    /// To support a graceful server shutdown we recommend using the `ServerQuiescingHelper` from the
+    /// SwiftNIO extras package. The `ServerQuiescingHelper` can be installed using the
+    /// ``ServerBootstrap/serverChannelInitializer`` callback.
+    ///
+    /// Below you can find the code to setup a simple TCP echo server that supports graceful server closure.
+    ///
+    /// ```swift
+    /// let quiesce = ServerQuiescingHelper(group: group)
+    /// let signalSource = DispatchSource.makeSignalSource(signal: SIGINT, queue: signalQueue)
+    /// signalSource.setEventHandler {
+    ///     signalSource.cancel()
+    ///     print("\nreceived signal, initiating shutdown which should complete after the last request finished.")
```

Comment:
NIT: We don't need the newline right?
```suggestion
    ///     print("received signal, initiating shutdown which should complete after the last request finished.")
```